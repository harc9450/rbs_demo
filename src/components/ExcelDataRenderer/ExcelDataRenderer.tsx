import React from "react";
import { IExcelDataToShow } from "../../modules/ExcelModules";

export const ExcelGrid = (props: IExcelDataToShow) => {
  const { rows, cols } = props;

  return (
    <div className="table table-responsive">
      <table className="table table-striped table-bordered">
        <tbody>
          {rows.map((value: any, index) => (
            <tr key={index}>
              {cols.map(column => {
                return column.name !== "A" && column.name !== "B" ? (
                  <td key={column.key}>{value[column.key]}</td>
                ) : null;
              })}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};
