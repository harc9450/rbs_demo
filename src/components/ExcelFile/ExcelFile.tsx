import * as React from "react";
import * as XLSX from "xlsx";
import { ExcelRenderer } from "../../helpers/ExcelHelper";
import { IExcelDataToShow } from "../../modules/ExcelModules";
import { ExcelGrid } from "../ExcelDataRenderer/ExcelDataRenderer";

export interface IExcelFileState {
  excelProps: IExcelDataToShow;
  isFileInvalid: boolean;
}
export interface IExcelFileProps {}

export default class ExcelFile extends React.Component<
  IExcelFileProps,
  IExcelFileState
> {
  constructor(props: IExcelFileProps) {
    super(props);

    this.state = {
      isFileInvalid: false,
      excelProps: {
        rows: [],
        cols: []
      }
    };
  }

  render(): JSX.Element {
    return (
      <div>
        <form>
          <div className="form-group">
            <input
              type="file"
              className="form-control btn btn-dark"
              onChange={this.fileChange}
            />
          </div>

          {!!this.state.isFileInvalid && (
            <div className="form-group text-danger">
              Please select a valid excel file
            </div>
          )}
        </form>

        <ExcelGrid {...this.state.excelProps} />
      </div>
    );
  }

  private fileChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    if (event.currentTarget.files && event.currentTarget.files.length > 0) {
      const file = event.currentTarget.files[0];
      const fileName = file.name;

      if (fileName.slice(fileName.lastIndexOf(".") + 1) === "xlsx") {
        this.setState({ isFileInvalid: false }, () => {
          this.handleFile(file);
        });
      } else {
        this.setState({
          isFileInvalid: true,
          excelProps: {
            rows: [],
            cols: []
          }
        });
      }
    }
  };

  private handleFile = (file: File): void => {
    ExcelRenderer(file).then((data: IExcelDataToShow) => {
      this.setState({ excelProps: data });
    });
  };
}
