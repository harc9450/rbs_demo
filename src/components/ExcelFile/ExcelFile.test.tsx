import ExcelFile, { IExcelFileProps } from "./ExcelFile";
import * as React from "react";
import Enzyme from "enzyme";
import toJson from "enzyme-to-json";
import { ExcelGrid } from "../ExcelDataRenderer/ExcelDataRenderer";

describe("ExcelFile Component", () => {
  let wrapper: Enzyme.ShallowWrapper;
  let mountedWrapper: Enzyme.ReactWrapper<any, any>;
  let props: IExcelFileProps;

  beforeEach(() => {
    props = {};
  });

  it("should render correctly", () => {
    wrapper = Enzyme.shallow(<ExcelFile {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should have a ExcelGrid element", () => {
    mountedWrapper = Enzyme.mount(<ExcelFile {...props} />);
    expect(mountedWrapper.find(ExcelGrid)).toHaveLength(1);
  });

  it("should have a form element", () => {
    mountedWrapper = Enzyme.mount(<ExcelFile {...props} />);
    expect(mountedWrapper.find("form")).toHaveLength(1);
  });

  it("should have a input element", () => {
    mountedWrapper = Enzyme.mount(<ExcelFile {...props} />);
    expect(mountedWrapper.find("input")).toHaveLength(1);
  });
});
