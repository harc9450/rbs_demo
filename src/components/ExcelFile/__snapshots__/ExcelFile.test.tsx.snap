// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render correctly 1`] = `
<div>
  <form>
    <div
      className="form-group"
    >
      <input
        className="form-control btn btn-dark"
        onChange={[Function]}
        type="file"
      />
    </div>
  </form>
  <ExcelGrid
    cols={Array []}
    rows={Array []}
  />
</div>
`;

exports[`ExcelFile Component should render correctly 1`] = `
<div>
  <form>
    <div
      className="form-group"
    >
      <input
        className="form-control btn btn-dark"
        onChange={[Function]}
        type="file"
      />
    </div>
  </form>
  <ExcelGrid
    cols={Array []}
    rows={Array []}
  />
</div>
`;
